echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -48,4 +49,26 @@ public static Boolean anyArtifactsMatchExpected(List<Artifact> messageArtifacts,
         .anyMatch(e -> e.matches(a));
     return messageArtifacts.stream().anyMatch(expectedArtifactMatch);
   }
+
+  /**
+   * Check that there is a key in the payload for each constraint declared in a Trigger.
+   * Also check that if there is a value for a given key, that the value matches the value in the payload.
+   * @param constraints A map of constraints configured in the Trigger (eg, created in Deck).
+   * @param payload A map of the payload contents POST'd in the Webhook.
+   * @return Whether every key (and value if applicable) in the constraints map is represented in the payload.
+   */
+  public static boolean isConstraintInPayload(final Map constraints, final Map payload) {
+    for (Object key : constraints.keySet()) {
+      if (!payload.containsKey(key) || payload.get(key) == null) {
+        log.info("Webhook trigger ignored. Item " + key.toString() + " was not found in payload");
+        return false;
+      }
+      if (!constraints.get(key).equals("") && (!constraints.get(key).equals(payload.get(key)))){
```

Comment:
Do we want this to `equals` or just `match`?

Reply:
Mmm match is a good idea